crates/engine/tree/src/tree/root.rs

```diff
@@ -673,6 +673,47 @@ where
         }
     }
 
+    /// Starts the main loop that handles all incoming messages, fetches proofs, applies them to the
+    /// sparse trie, updates the sparse trie, and eventually returns the state root.
+    ///
+    /// The lifecycle is the following:
+    /// 1. Either [`StateRootMessage::PrefetchProofs`] or [`StateRootMessage::StateUpdate`] is
```

Comment:
can we add better comment on the `StateRootMessage::PrefetchProofs` , it's not immediately obvious what the message means